src/server.h

```diff
@@ -1292,6 +1292,7 @@ typedef struct redisOp {
 typedef struct redisOpArray {
     redisOp *ops;
     int numops;
+    int len;
```

Comment:
what's the difference between `len` and `numops`?

Reply:
i mean, i can see that `len` is the number of "taken" elements and `numops` is the number of taken elements that have a `target` (i.e not the "placeholder" ones) - but do we really need it?

the placeholder must be filled before calling `propagatePendingCommands` (we should assert that `target` exists) so maybe we can drop `len` and use only `numops`, which will be incremented, regardless of whether it's a placeholder or not

Reply:
The placeholder is not necessarily be set in case the command did not cause any effects and does not need to be replicated.